### PR TITLE
refactor etcd to meet the chart tech requirement

### PR DESCRIPTION
**Pull request by cheyang**
**What this PR does / why we need it**:

1. de-capitalize variables in etcd
2. make the image pull policy as "IfNotPresent"
3. Use image.repository to replace image

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: incubator/etcd/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 name: etcd
 home: https://github.com/coreos/etcd
-version: 0.3.9
+version: 0.3.10
```
Comment (unguiculus) [inline incubator/etcd/Chart.yaml]: This should be a major bump. Once this is through, we should move the chart to stable.

**Review by unguiculus — CHANGES_REQUESTED** [review]

**Comment by unguiculus** [comment]
So, let's go with a minor bump for now as long as it is in incubator. Moving it to stable would warrant a 1.0.0.

**Review by unguiculus — APPROVED** [review]

**Comment by cheyang** [comment]
@unguiculus I noticed the reason of the e2e testing failed due to the following error:

```
Events:
  Type     Reason            Age               From               Message
  ----     ------            ----              ----               -------
  Warning  FailedScheduling  11s (x8 over 1m)  default-scheduler  PersistentVolumeClaim is not bound: "datadir-etcd-etcd-0" (repeated 6 times)
```
My solution is adding `if .Values.storage` to determine if it needs volumeClaimTemplates:
```
{{- if .Values.storage }}
  volumeClaimTemplates:
  ...
  {{- end }}
```

**Comment by unguiculus** [comment]
CI had problems with PVCs not getting bound.

**Comment by cheyang** [comment]
Do you think my change of not using pvc by default is fine?

**Comment by unguiculus** [comment]
@cheyang Please revert that change. You could make persistence optional, but please follow the standard approach and provide an emptyDir instead.

Here's an example:
https://github.com/kubernetes/charts/blob/master/stable/mongodb-replicaset/templates/mongodb-statefulset.yaml#L218-L244

**Comment by cheyang** [comment]
@unguiculus pls review when you have time, thx.

File: incubator/etcd/values.yaml [hunk 2]
```diff
@@ -3,29 +3,32 @@
...
-##
-# storageClass: "-"
+peerPort: 2380
+clientPort: 2379
+component: "etcd"
+replicas: 3
+image:
+  repository: "k8s.gcr.io/etcd-amd64"
+  tag: "2.2.5"
+  pullPolicy: "Always"
```
Comment (unguiculus) [inline incubator/etcd/values.yaml]: IfNotPresent

File: incubator/etcd/templates/service.yaml [hunk 3]
```diff
@@ -9,13 +9,13 @@ metadata:
     heritage: {{ .Release.Service | quote }}
     release: {{ .Release.Name | quote }}
     chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
-    component: "{{ .Release.Name }}-{{ .Values.Component }}"
+    component: "{{ .Release.Name }}-{{ .Values.component }}"
```
Comment (unguiculus) [inline incubator/etcd/templates/service.yaml]: The chart doesn't have multiple components. Why not remove the label? Instead you should add the `app` label.

File: incubator/etcd/templates/statefulset.yaml [hunk 4]
```diff
@@ -6,47 +6,47 @@ metadata:
     heritage: {{ .Release.Service | quote }}
     release: {{ .Release.Name | quote }}
     chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
-    component: "{{ .Release.Name }}-{{ .Values.Component }}"
+    component: "{{ .Release.Name }}-{{ .Values.component }}"
 spec:
   serviceName: {{ template "etcd.fullname" . }}
-  replicas: {{ default 3 .Values.Replicas }}
+  replicas: {{ default 3 .Values.replicas }}
```
Comment (unguiculus) [inline incubator/etcd/templates/statefulset.yaml]: Remove the default funtion. It's redundant.

**Review by unguiculus — CHANGES_REQUESTED** [review]
Please update `_helpers.tpl` as well.

File: incubator/etcd/README.md [hunk 5]
```diff
@@ -95,7 +94,7 @@ $ kill -9 ETCD_1_PID
 ```
 
 ```shell
-$ kubectl get pods -l "component=${RELEASE-NAME}-etcd"
+$ kubectl get pods -l "release=${RELEASE-NAME}"
```
Comment (unguiculus) [inline incubator/etcd/README.md]: You also need to add the `app` label to selectors.

File: incubator/etcd/templates/statefulset.yaml [hunk 6]
```diff
@@ -6,47 +6,47 @@ metadata:
...
         resources:
           requests:
-            cpu: "{{ .Values.Cpu }}"
-            memory: "{{ .Values.Memory }}"
+            cpu: "{{ .Values.cpu }}"
+            memory: "{{ .Values.memory }}"
         env:
         - name: INITIAL_CLUSTER_SIZE
-          value: {{ default 3 .Values.Replicas | quote }}
+          value: {{ default 3 .Values.replicas | quote }}
```
Comment (unguiculus) [inline incubator/etcd/templates/statefulset.yaml]: Don't use the `default` function if not really necessary. Defaults should come from `values.yaml`.

File: incubator/etcd/templates/statefulset.yaml [hunk 7]
```diff
@@ -6,47 +6,47 @@ metadata:
...
+        imagePullPolicy: "{{ .Values.image.pullPolicy }}"
         ports:
-        - containerPort: {{ .Values.PeerPort }}
+        - containerPort: {{ .Values.peerPort }}
           name: peer
-        - containerPort: {{ .Values.ClientPort }}
+        - containerPort: {{ .Values.clientPort }}
           name: client
         resources:
           requests:
```
Comment (unguiculus) [inline incubator/etcd/templates/statefulset.yaml]: Don't specify defaults for resources. Just do this:

`{{ toYaml .Values.resources | indent 10 }}`

**Review by unguiculus — CHANGES_REQUESTED** [review]
A few more things.

File: incubator/etcd/templates/statefulset.yaml [hunk 8]
```diff
@@ -6,47 +6,45 @@ metadata:
     heritage: {{ .Release.Service | quote }}
     release: {{ .Release.Name | quote }}
     chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
-    component: "{{ .Release.Name }}-{{ .Values.Component }}"
+    app: {{ template "etcd.name" . }}
 spec:
   serviceName: {{ template "etcd.fullname" . }}
-  replicas: {{ default 3 .Values.Replicas }}
+  replicas: {{ default 3 .Values.replicas }}
```
Comment (unguiculus) [inline incubator/etcd/templates/statefulset.yaml]: Remove `default` function. Please check again and apply my comments to the whole chart.

**Review by unguiculus — APPROVED** [review]

**Pull request merged**